### ISSUE TITLE
Directory exclusion is case-sensitive: Build/, Dist/, Temp/, Logs/ leak into the tree and analysis

### ISSUE THREAD

**Issue by wallidsaydi-creator** [trevor-nichols/agentrules-architect#39]
## Summary

Directory exclusion in `should_exclude()` (src/agentrules/core/utils/file_system/tree_generator.py) matches directory names **case-sensitively**, while file patterns in the same function match **case-insensitively**. Capitalized build/artifact directories therefore leak into the "exclusion-aware" tree — and into the analyze phase's file retrieval, since `list_files()` calls the same `should_exclude()`.

Real-world casing that leaks: Unity and .NET produce `Build/`, npm/Vite configs commonly use `Dist/`, Windows tooling often creates `Temp/`, `Logs/`.

## Repro

```bash
mkdir -p /tmp/ara-case/{src,Dist,Build,Logs,Tmp,node_modules,readme-fix}
echo x > /tmp/ara-case/src/app.py
echo x > /tmp/ara-case/Dist/bundle.js
echo x > /tmp/ara-case/Build/out.js
echo x > /tmp/ara-case/Logs/app.log
echo x > /tmp/ara-case/Tmp/temp.js
echo x > /tmp/ara-case/node_modules/pkg.js
echo x > /tmp/ara-case/readme-fix/note.md

agentrules tree /tmp/ara-case --max-depth 3
```

**Before (v0.7.0, current main):** `Build/`, `Dist/`, `Tmp/` and their contents all appear in the tree:

```
<project_structure>
├── Build
│   └── out.js
├── Dist
│   └── bundle.js
├── Logs          <- empty only because *.log is caught by the (case-insensitive) file-pattern check
├── readme-fix
│   └── note.md
├── src
│   └── app.py
└── Tmp
    └── temp.js
</project_structure>
```

Note the asymmetry inside one function: `Logs/app.log` *was* excluded because file patterns go through `fnmatch(item.name.lower(), pattern.lower())`, but the directory check `item.name in exclude_dirs` keeps original casing. The dead giveaway in the source:

```python
normalized_name = item.name.lower()   # computed...
if item.name in exclude_dirs:         # ...but not used for the dir check
```

## Why it matters beyond the tree

`list_files()` in file_retriever.py calls the same `should_exclude()`, and `get_file_contents()` reads up to `max_files=100` files into the analysis. In a Unity/.NET-style repo, leaked `Build/` artifacts crowd real source out of the LLM analysis budget and can put build-generated facts into the generated AGENTS.md/CLAUDE.md — the exact docs-drift class this tool exists to prevent.

## Suggested fix (1 line)

diff --git a/src/agentrules/core/utils/file_system/tree_generator.py b/src/agentrules/core/utils/file_system/tree_generator.py
index dc12e11..6fb0b20 100644
--- a/src/agentrules/core/utils/file_system/tree_generator.py
+++ b/src/agentrules/core/utils/file_system/tree_generator.py
@@ -75,7 +75,7 @@ def should_exclude(item: Path, exclude_dirs: set[str], exclude_patterns: set[str
     # Check if it's a directory in the exclude list
     if item.is_dir():
         normalized_name = item.name.lower()
-        if item.name in exclude_dirs:
+        if item.name in exclude_dirs or normalized_name in {d.lower() for d in exclude_dirs}:
             return True
         if normalized_name.startswith('.') and normalized_name.endswith('_cache'):
             return True

Casefolding the comparison only; no behavior change for already-lowercase names. After the patch, the same fixture produces:

```
<project_structure>
├── readme-fix
│   └── note.md
└── src
    └── app.py
</project_structure>
```

## Verification

- Repro above on clean main: `Build/`, `Dist/`, `Tmp/` leak (13 tree lines).
- With the 1-line patch: 6 tree lines, all three artifact dirs excluded, `src/` and `readme-fix/` intact.
- Test suite with patch applied: 885 passed, 10 skipped (2 failures in `tests/unit/agents/test_anthropic_client_compat.py` are pre-existing on unpatched main — verified via `git stash` before/after: same 2 failed, 7 passed in that file).

---

*Found while running a docs-vs-code drift check on the repo for [FreshContext Pack](https://deploy-foorge-team.vercel.app/) — a $5 repo-specific freshness add-on for AI coding agents. If useful, happy to open a PR with this fix + a regression test.*
